### PR TITLE
fix: use updated deployNewProxy fn

### PR DESCRIPTION
**Pull request by james-a-morris**
The current references to `deployNewProxy` result in compilation errors. We should modify `deployNewProxy` so that it can pass an initializer & constructor args and call that fn correctly.

### CODE REVIEW

**Review by nicholaspai — APPROVED** [review]

File: deploy/003_deploy_optimism_spokepool.ts [hunk 1]
```diff
@@ -9,9 +9,7 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   // with deprecated spoke pool.
   // Set hub pool as cross domain admin since it delegatecalls the Adapter logic.
   const initArgs = [1_000_000, hubPool.address, hubPool.address];
-  await deployNewProxy("Optimism_SpokePool", initArgs, {
-    constructorArgs: ["0x4200000000000000000000000000000000000006", 3600, 32400],
-  });
+  await deployNewProxy("Optimism_SpokePool", ["0x4200000000000000000000000000000000000006", 3600, 32400], initArgs);
```
Comment (pxrl) [inline deploy/003_deploy_optimism_spokepool.ts:12]: It'd be nice to explicitly call out the constructor args -wdyt?

```suggestion
  const constructorArgs = ["0x4200000000000000000000000000000000000006", 3600, 32400];
  await deployNewProxy("Optimism_SpokePool", constructorArgs, initArgs);
```

File: utils/utils.hre.ts [hunk 2]
```diff
@@ -23,12 +23,15 @@ export async function getSpokePoolDeploymentInfo(
   return { hubPool, hubChainId, spokeChainId };
 }
 
-export async function deployNewProxy(name: string, args: (number | string)[]): Promise<void> {
+type FnArgs = number | string;
+export async function deployNewProxy(name: string, constructorArgs: FnArgs[], initArgs: FnArgs[]): Promise<void> {
```
Comment (pxrl) [inline utils/utils.hre.ts:27]: Nice to name it as it is used in `run("verify:verify": ...);`.

```suggestion
export async function deployNewProxy(name: string, constructorArguments: FnArgs[], initArgs: FnArgs[]): Promise<void> {
```

**Review by pxrl — APPROVED** [review]
Couple of minor comments.

File: deploy/003_deploy_optimism_spokepool.ts (hunk 1 above)
Comment (james-a-morris) [reply deploy/003_deploy_optimism_spokepool.ts:12]: This was an existing pattern in the code. Do you think it should be changed in a separate PR?

File: utils/utils.hre.ts (hunk 2 above)
Comment (james-a-morris) [reply utils/utils.hre.ts:27]: Fair but this is also used slightly further down as `constructorArgs` in the `deployProxy` fn. 

```
  const proxy = await upgrades.deployProxy(await getContractFactory(name, {}), initArgs, {
    kind: "uups",
    unsafeAllow: ["delegatecall"], // Remove after upgrading openzeppelin-contracts-upgradeable post v4.9.3.
    constructorArgs,
    initializer: "initialize",
  });
```

File: deploy/003_deploy_optimism_spokepool.ts (hunk 1 above)
Comment (pxrl) [reply deploy/003_deploy_optimism_spokepool.ts:12]: It's directly related to this change so I think we should just fix it in place :+1:

File: utils/utils.hre.ts (hunk 2 above)
Comment (pxrl) [reply utils/utils.hre.ts:27]: Ah - that's annoying. Ignore this one then :+1:

File: deploy/003_deploy_optimism_spokepool.ts (hunk 1 above)
Comment (james-a-morris) [reply deploy/003_deploy_optimism_spokepool.ts]: Done

File: deploy/011_deploy_polygon_spokepool.ts [hunk 3]
```diff
@@ -18,9 +18,8 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     hubPool.address,
     L2_ADDRESS_MAP[spokeChainId].fxChild,
   ];
-  await deployNewProxy("Polygon_SpokePool", initArgs, {
-    constructorArgs: [L2_ADDRESS_MAP[spokeChainId].wMatic, 3600, 32400],
-  });
+  const constructorArgs = [L2_ADDRESS_MAP[spokeChainId].wMatic, 3600, 32400];
+  await deployNewProxy("Polygon_SpokePool", constructorArgs, initArgs);
```
Comment (pxrl) [inline deploy/011_deploy_polygon_spokepool.ts:22]: Is the order of `constructorArgs` vs. `initArgs` correct on this one? It looks like they're now reversed vs. previous, and `consutrctorArgs` used to be supplied as part of a larger object.

Comment (pxrl) [reply deploy/011_deploy_polygon_spokepool.ts:22]: Nevermind - I see that the prototype for deployNewProxy has also been updated :+1:

Comment (james-a-morris) [reply deploy/011_deploy_polygon_spokepool.ts:22]: The Polygon Spoke's constructor takes 3 items:
```
constructor(
        address _wrappedNativeTokenAddress,
        uint32 _depositQuoteTimeBuffer,
        uint32 _fillDeadlineBuffer
    )
```

The `initialize` function (`initArgs`) takes 5:
```
function initialize(
        uint32 _initialDepositId,
        PolygonTokenBridger _polygonTokenBridger,
        address _crossDomainAdmin,
        address _hubPool,
        address _fxChild
    )
```

**Pull request merged**